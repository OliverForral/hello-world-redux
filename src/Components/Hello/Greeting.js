import React from 'react';
import PropTypes from 'prop-types';

const Greeting = ({name}) => (
  <div>{name && `Hello, ${name}!`}</div>
)

Greeting.propTypes = {
  name: PropTypes.string,
};

export default Greeting;
