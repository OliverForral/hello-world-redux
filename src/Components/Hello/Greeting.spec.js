import React from 'react';
import renderer from 'react-test-renderer';
import Greeting from './Greeting';

it('renders correctly', () => {
  const tree = renderer
    .create(<Greeting />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when given a name', () => {
  const tree = renderer
    .create(<Greeting name="world" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
