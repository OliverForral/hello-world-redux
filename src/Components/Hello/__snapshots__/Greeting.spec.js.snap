// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `<div />`;

exports[`renders correctly when given a name 1`] = `
<div>
  Hello, world!
</div>
`;
